velox/expression/CastExpr-inl.h

```diff
@@ -327,37 +340,39 @@ VectorPtr CastExpr::applyDecimalToIntegralCast(
   const auto precisionScale = getDecimalPrecisionScale(*fromType);
   const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
   const auto scaleFactor = DecimalUtil::kPowersOfTen[precisionScale.second];
-  const auto castToIntByTruncate =
-      context.execCtx()->queryCtx()->queryConfig().isCastToIntByTruncate();
-  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
-    auto value = simpleInput->valueAt(row);
-    auto integralPart = value / scaleFactor;
-    if (!castToIntByTruncate) {
+  if (hooks_->truncate()) {
+    rows.applyToSelected([&](auto row) {
```

Comment:
Hi @rui-mo, Sorry, just noticed this piece. Shall we use `applyToSelectedNoThrowLocal` instead of rows.applyToSelected since simpleInput->valueAt(row) could throw in debug mode?

Reply:
Thanks for catching. Updated.